Modules/syslogmodule.c

```diff
@@ -191,8 +191,11 @@ syslog_syslog(PyObject * self, PyObject * args)
          */
         if ((openargs = PyTuple_New(0))) {
             PyObject *openlog_ret = syslog_openlog(self, openargs, NULL);
-            Py_XDECREF(openlog_ret);
             Py_DECREF(openargs);
+            Py_XDECREF(openlog_ret);
+            if (openlog_ret == NULL) {
+                return NULL;
+            }
```

Comment:
Technically, comparing a pointer to freed memory is an undefined behavior. You should compare it before decrefing.